File: zul/src/org/zkoss/zul/Space.java
/* Space.java

	Purpose:
		
	Description:
		
	History:
		Sat Jan 14 00:40:57     2006, Created by tomyeh

Copyright (C) 2006 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under LGPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
*/
package org.zkoss.zul;

import org.zkoss.zk.ui.WrongValueException;

/**
 * Space is a {@link Separator} with the orient default to "vertical".
 *
 * <p>In other words, &lt;space&gt; is equivalent to &lt;separator orient="vertical"&gt;
 *
 * @author tomyeh
 */
public class Space extends Separator implements org.zkoss.zul.api.Space {
	public Space() {
		try {
			setOrient("vertical");
		} catch (WrongValueException ex) {
			throw new InternalError(); //impossible
		}
	}
}

File: zul/src/main/resources/web/js/zul/wgt/Space.ts
/* Space.js

	Purpose:

	Description:

	History:
		Mon Nov 17 09:33:07     2008, Created by tomyeh

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
/**
 * Space is a {@link Separator} with the orient default to "vertical".
 */
@zk.WrapClass('zul.wgt.Space')
export class Space extends zul.wgt.Separator {
	/** @internal */
	override _orient = 'vertical';
}